File: snippets/dotdict/dotdict.py
initials = {}
for name in ('peter', 'anders', 'bengt', 'bengtsson'):
    initial = name[0]
    # if initial not in initials:
    #     initials[initial] = 0
    initials.setdefault(initial, 0)
    initials[initial] += 1
print initials
# outputs
# {'a': 1, 'p': 1, 'b': 2}

File: snippets/dotdict/dotdict.go
package main

import "fmt"

func main() {
	names := []string{"peter", "anders", "bengt", "bengtsson"}
	initials := make(map[string]int)
	for _, name := range names {
		initial := string(name[0])
		initials[initial]++
	}
	fmt.Println(initials)
	// outputs
	// map[p:1 a:1 b:2]
}
